### PR TITLE
Add pop/push size per instruction

### PR DESCRIPTION
**Pull request by montyly**

### CODE REVIEW

**Comment by montyly** [comment]
TODO: improve testing coverage so that we cover the new code

File: tealer/teal/instructions/instructions.py [hunk 1]
```diff
@@ -333,6 +349,10 @@ def value(self) -> Union[str, int]:
     def __str__(self) -> str:
         return f"pushint {self._value}"
 
+    @property
+    def stack_push_size(self) -> int:
+        return 1
+
 
 class Txn(Instruction):
```
Comment (S3v3ru5) [inline tealer/teal/instructions/instructions.py:357]: Add `stack_push_size = 1` for Txn instruction

File: tealer/teal/instructions/instructions.py [hunk 2]
```diff
@@ -1173,6 +1405,14 @@ def cost(self) -> int:
...
+    @property
+    def stack_pop_size(self) -> int:
+        return 4
+
+    @property
+    def stack_push_size(self) -> int:
+        return 2
+
 
 class Global(Instruction):
```
Comment (S3v3ru5) [inline tealer/teal/instructions/instructions.py:1417]: Add `stack_push_size = 1` for Global Instruction

File: tealer/teal/instructions/instructions.py [hunk 3]
```diff
@@ -1228,6 +1484,14 @@ def __init__(self) -> None:
...
+    @property
+    def stack_pop_size(self) -> int:
+        return 3
+
+    @property
+    def stack_push_size(self) -> int:
+        return 1
+
 
 class Cover(Instruction):
```
Comment (S3v3ru5) [inline tealer/teal/instructions/instructions.py:1496]: Add `stack_pop_size = 1 + self._idx` and `stack_push_size = 1 + self._idx` to cover instruction (?)

File: tealer/teal/instructions/instructions.py [hunk 4]
```diff
@@ -1444,6 +1736,10 @@ def __init__(self) -> None:
     def __str__(self) -> str:
```
Comment (S3v3ru5) [inline tealer/teal/instructions/instructions.py:1736]: Add `stack_push_size = 1` for AppGlobalGet (app_global_get) instruction.

File: tealer/teal/instructions/instructions.py [hunk 5]
```diff
@@ -1820,6 +2212,14 @@ class Btoi(Instruction):
...
+    @property
+    def stack_pop_size(self) -> int:
+        return 1
+
+    @property
+    def stack_push_size(self) -> int:
+        return 1
+
 
 class Addr(Instruction):
```
Comment (S3v3ru5) [inline tealer/teal/instructions/instructions.py:2224]: add `stack_push_size = 1` for Addr instruction.

File: tealer/teal/instructions/instructions.py [hunk 6]
```diff
@@ -2678,6 +3346,14 @@ def cost(self) -> int:
             return 4
         return 0
 
+    @property
+    def stack_pop_size(self) -> int:
+        return 2
```
Comment (S3v3ru5) [inline tealer/teal/instructions/instructions.py:3351]: Update to `stack_pop_size = 1` for BBitwiseInvert (b~) instruction

File: tealer/teal/instructions/instructions.py [hunk 7]
```diff
@@ -3238,6 +4046,10 @@ class Intc2(Instruction):
     def __str__(self) -> str:
         return "intc_2"
 
+    @property
+    def stack_push_size(self) -> int:
+        return 1
+
 
 class Intc3(Instruction):
```
Comment (S3v3ru5) [inline tealer/teal/instructions/instructions.py:4054]: Add `stack_push_size = 3` for Intc3

File: tealer/teal/instructions/instructions.py [hunk 8]
```diff
@@ -3509,6 +4373,14 @@ def __init__(self, start: int, stop: int):
...
+    @property
+    def stack_pop_size(self) -> int:
+        return 1
+
+    @property
+    def stack_push_size(self) -> int:
+        return 1
+
 
 class Substring3(Instruction):
```
Comment (S3v3ru5) [inline tealer/teal/instructions/instructions.py:4385]: Add `stack_push_size = 1` and `stack_pop_size = 3` to Substring3 Instruction

File: tealer/teal/instructions/instructions.py [hunk 9]
```diff
@@ -3760,6 +4676,10 @@ def field(self) -> TransactionField:
         """Array transaction field being accessed."""
         return self._field
```
Comment (S3v3ru5) [inline tealer/teal/instructions/instructions.py:4678]: Add `stack_pop_size = 1` to Itxnas instruction

File: tealer/teal/instructions/instructions.py [hunk 10]
```diff
@@ -3903,6 +4847,14 @@ def start_position(self) -> int:
             raise TealerException("replace instruction does not have any immediates")
         return self._idx
 
+    @property
```
Comment (S3v3ru5) [inline tealer/teal/instructions/instructions.py:4850]: Number of values popped depends on the immediate argument `s`. If there's **no** immediate argument then, `replace` is `replace3` and `stack_pop_size = 3`. If there is a immediate argument then `replace s` is equivalent to `replace2 s` and `stack_pop_size = 2`.

File: tealer/teal/instructions/instructions.py [hunk 11]
```diff
@@ -4164,6 +5172,10 @@ def __init__(self, n: int):
         self._nelements: int = n
         self._version: int = 8
 
+    @property
```
Comment (S3v3ru5) [inline tealer/teal/instructions/instructions.py:5175]: Update `stack_pop_size = n` for Popn instruction

File: tealer/teal/instructions/instructions.py [hunk 12]
```diff
@@ -4183,6 +5195,10 @@ def __init__(self, count: int):
         self._count: int = count
         self._version: int = 8
 
+    @property
```
Comment (S3v3ru5) [inline tealer/teal/instructions/instructions.py:5198]: Update `stack_push_size = n + 1` and `stack_pop_size = 1` for Dupn instruction

File: tealer/teal/instructions/instructions.py [hunk 13]
```diff
@@ -4204,6 +5220,10 @@ def __init__(self, bytes_list: List[str]):
         self._bytes_list = bytes_list
         self._version: int = 8
 
+    @property
+    def stack_push_size(self) -> int:
+        return 1
```
Comment (S3v3ru5) [inline tealer/teal/instructions/instructions.py:5225]: update `stack_push_size = len(self._bytes_list)` for PushBytess instruction.

**Pull request merged**